sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.students.controller.getOfferingSummaryDetails.e2e-spec.ts

```diff
@@ -0,0 +1,183 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getStudentToken,
+  getStudentByFakeStudentUserType,
+} from "../../../../testHelpers";
+import {
+  createE2EDataSources,
+  createFakeApplication,
+  createFakeEducationProgramOffering,
+  createFakeStudentAssessment,
+  saveFakeApplicationOfferingRequestChange,
+  createFakeUser,
+  E2EDataSources,
+} from "@sims/test-utils";
+import {
+  Application,
+  StudentAssessment,
+  Student,
+  EducationProgramOffering,
+} from "@sims/sims-db";
+import { Repository } from "typeorm";
+
+describe("EducationProgramOfferingStudentsController(e2e)-getOfferingSummaryDetails", () => {
+  let app: INestApplication;
+  let student: Student;
+  let db: E2EDataSources;
+  let applicationRepo: Repository<Application>;
+  let assessmentRepo: Repository<StudentAssessment>;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    applicationRepo = dataSource.getRepository(Application);
+    assessmentRepo = dataSource.getRepository(StudentAssessment);
+    student = await getStudentByFakeStudentUserType(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+      dataSource,
+    );
+  });
+
+  it("Should get the offering simplified details, not including the validations, approvals and extensive data when the provided purpose is application-offering-change.", async () => {
+    // Arrange
+    const savedUser = await db.user.save(createFakeUser());
+    const fakeOffering = createFakeEducationProgramOffering({
+      auditUser: savedUser,
+    });
+    const offering = await db.educationProgramOffering.save(fakeOffering);
+    const fakeApplication = createFakeApplication({ student });
+    const application = await applicationRepo.save(fakeApplication);
+    const fakeAssessment = createFakeStudentAssessment({
+      auditUser: savedUser,
+      application,
+      offering,
+    });
+    fakeApplication.currentAssessment = fakeAssessment;
+    await applicationRepo.save(fakeApplication);
+    await assessmentRepo.save(fakeAssessment);
```

Comment:
We could have used the saveFakeApplication with student and it would have replaced all the code here right?

I am not going to call it a blocker but to keep in mind for upcoming e2es.

Reply:
I have made the changes @dheepak-aot . Thanks for pointing it out. 👍